### PR TITLE
Scanners

### PR DESCRIPTION
**Pull request by Ethanljf**
+ Initial layout of knative scanning structure

### CODE REVIEW

File: api/app/logger.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                'graphql_located_error_filter',
+                'graphql_traceback_error_filter',
+            ],
+        },
+    },
+}
+
+logging.config.dictConfig(logger_dict)
+logger = logging.getLogger('custom')
+# logging.error('Error message')
```
Comment (peacheym) [inline api/app/logger.py:65]: Consider removing commented out line.

File: platform/overlays/knative/images/dispatcher/dispatcher.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+
+@app.route('/receive', methods=['GET', 'POST'])
+def receive():
+
+    payload = {}
+    dkim_flag = False
+
+    try:
+        #TODO Replace secret
```
Comment (peacheym) [inline platform/overlays/knative/images/dispatcher/dispatcher.py:39]: Does this TODO need to be resolved?  If already done, consider removing comment.

File: platform/overlays/knative/images/dispatcher/dispatcher.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        )
+
+        test_flag = request.headers.get("Test")
+
+        payload = {"scan_id": decoded_payload["scan_id"], "domain": decoded_payload["domain"]}
+        dkim_flag = decoded_payload["dkim"]
+        user_initialized = decoded_payload["user_init"]
+        scan_id = decoded_payload["scan_id"]
+
+        # TODO Replace secret
```
Comment (peacheym) [inline platform/overlays/knative/images/dispatcher/dispatcher.py:53]: Same issue here!  Consider removing comment or resolving TODO item.

File: platform/overlays/knative/images/dkim/dkim_scanner.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+app = Flask(__name__)
+
+
+@app.route('/receive', methods=['GET', 'POST'])
+def receive():
+
+    logging.info("Event received\n")
+
+    try:
+        # TODO Replace secret
```
Comment (peacheym) [inline platform/overlays/knative/images/dkim/dkim_scanner.py:29]: TODO item?

File: platform/overlays/knative/images/dkim/dkim_scanner.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+    except Exception as e:
+        logging.error(str(e)+'\n')
+        return 'Failed to send scan to result-handling service'
+
+
+def dispatch(scan_id, payload):
+    try:
+        # Post request to result-handling service
+        # TODO: Pull values from env
```
Comment (peacheym) [inline platform/overlays/knative/images/dkim/dkim_scanner.py:76]: consider address this todo item or removing the comment if already resolved.

File: platform/overlays/knative/images/dmarc/dmarc_scanner.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+app = Flask(__name__)
+
+
+@app.route('/receive', methods=['GET', 'POST'])
+def receive():
+
+    logging.info("Event received\n")
+
+    try:
+        # TODO Replace secret
```
Comment (peacheym) [inline platform/overlays/knative/images/dmarc/dmarc_scanner.py:28]: Todo item?

File: platform/overlays/knative/images/dmarc/dmarc_scanner.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    except Exception as e:
+        logging.error(str(e)+'\n')
+        return 'Failed to send scan to result-handling service'
+
+
+def dispatch(scan_id, payload):
+    try:
+        # Post request to result-handling service
+        # TODO: Pull values from env
+        response = requests.post('http://34.67.57.19/receive', headers=headers, data=payload)
```
Comment (peacheym) [inline platform/overlays/knative/images/dmarc/dmarc_scanner.py:78]: Todo Item, looks like it needs an ENV ! :smile:

File: .editorconfig [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# http://editorconfig.org
```
[the remark this answers is not in the document]
Comment (Ethanljf) [reply .editorconfig]: As it's included in master, I will include the file and remove it from .gitignore

File: platform/overlays/knative/images/dispatcher/dispatcher.py (hunk 2 above)
Comment (Ethanljf) [reply platform/overlays/knative/images/dispatcher/dispatcher.py:39]: Resolved!

**Pull request closed without merging**